e2e/scripts/mrt-target-manager.js

```diff
@@ -0,0 +1,322 @@
+
+class MRTTargetManager {
+    constructor(options = {}) {
+        this.bucket = options.bucket
+        this.poolDataFileKey = options.poolDataFileKey
+        this.maxRetries = options.maxRetries || 3
+        this.retryDelay = options.retryDelay || 10000 // 10 seconds
+        this.prNumber = options.prNumber || process.env.GITHUB_PR_NUMBER || null
+        this.branch = options.branch || null
+        this.runId = options.runId || null
+        this.s3Client = new SecureS3Client({
+            region: options.region,
+            readOnly: !process.env.CI,
+            roleArn: process.env.CI ? options.roleArn : options.roleArn, // Don't use role ARN in CI since AWS credentials action handles it
+            roleSessionName: options.roleSessionName || 'LocalDev'
+        })
+    }
+
+    async initialize() {
+        await this.s3Client.initialize()
+    }
+
+    /**
+     * Convert a ReadableStream object returned by the S3 client to a string
+     * @param {Stream} stream - The stream to convert
+     * @returns {Promise<string>} - The string representation of the stream
+     */
+    async streamToString(stream) {
+        const chunks = []
+        for await (const chunk of stream) {
+            chunks.push(chunk)
+        }
+        return Buffer.concat(chunks).toString()
+    }
+
+    /**
+     * Download the pool file and return data with ETag
+     */
+    async downloadPoolFile() {
+        try {
+            const downloadResult = await this.s3Client.download(this.bucket, this.poolDataFileKey)
+            // Convert stream to string and parse JSON
+            const contentString = await this.streamToString(downloadResult.body)
+            const poolData = JSON.parse(contentString)
+
+            return {
+                ...downloadResult,
+                poolData
+            }
+        } catch (error) {
+            if (error.name === 'NoSuchKey') {
+                console.log('❌ Pool file not found.')
+            }
+            throw error
+        }
+    }
+
+    /**
+     * Find an available environment of the specified type
+     */
+    findAvailableEnvironment(poolData) {
+        const availableEnvs = poolData.environments.filter(
+            (env) => env.ciAvailability === 'available'
+        )
+
+        if (availableEnvs.length === 0) {
+            return null
+        }
+
+        return availableEnvs[0]
+    }
+
+    /**
+     * Mark environment as in-use by current PR
+     */
+    updateMRTTargetStatus(poolData, environment, ciAvailability) {
+        const updatedPoolData = {
+            ...poolData,
+            environments: poolData.environments.map((env) => {
+                if (env.slug === environment.slug) {
+                    const updatedEnv = {
+                        ...env,
+                        ciAvailability,
+                        ciAcquiredAt: new Date().toISOString(),
+                        ciLastUsed: new Date().toISOString()
```

Comment:
Do you think it is better to group ci-related infor into it owns `ci` field

```

 const updatedEnv = {
                        ...env,
                        ci: {
                            available: true,
                            acquiredAt: new Date().toISOString()
                        
                        }

}


```

Reply:
I did that for fields which are specific to CI runs and which will be deleted when the run ends.
I've left `ciAvailability` and `ciLastUsed` at the top level since they're more contextual to give information about the MRT target rather than the CI run.

Here's the updated structure:
```
{
      "slug": "stg-001",
      "name": "stg-001",
      "hostname": null,
      "ssrExternalHostname": "pwa-kit-ci-stg-001.mrt-storefront-staging.com",
      "ssrExternalDomain": "mrt-storefront-staging.com",
      "ssrRegion": "us-east-2",
      "ciAvailability": "in-use",
      "ciLastUsed": "2025-08-06T00:28:08.231Z",
      "ciRunInfo": {
        "ciAcquiredAt": "2025-08-06T00:28:08.231Z",
        "branch": "feature/demo-feature",
        "runId": "123"
      }
}
```